click_repl/__init__.py

```diff
@@ -135,6 +140,13 @@ def repl(old_ctx, prompt_kwargs=None):
     group = group_ctx.command
     isatty = sys.stdin.isatty()
 
+    # Delete the REPL command from those available, as we don't want to allow
+    # nesting REPLs (note: pass `None` to `pop` as we don't want to error if
+    # REPL command already not present for some reason).
+    repl_command_name = old_ctx.command.name
+    available_commands = group_ctx.command.commands
+    available_commands.pop(repl_command_name, None)
```

Comment:
I'd much prefer if we passed the repl command name into `ClickCompleter` such that it can do the filtering without mutating the original commands.

Then we can just pass the REPL name to `ClickCompleter` in the tests as well and check if it's filtered.